velox/exec/MergeJoin.h

```diff
@@ -360,10 +360,17 @@ class MergeJoin : public Operator {
   // Join filter input type.
   RowTypePtr filterInputType_;
 
-  // Maps left-side input channels to channels in 'filterInputType_'.
+  // Maps 'filterInputType_' channels to the corresponding channels in output_,
```

Comment:
Wouldn't it be more convenient to map output columns to filter input columns?

Also, wondering it makes sense to store the mapping as a vector. Let's say we have 10 output columns and 3 of them are also used in the filter. We can have a vector like this:

[-1, -1, 2, -1, 0, -1, 1, -1, -1, -1]

which tells use that output columns 3, 5 and 7 can be fetched from the filter input columns 1, 2, 3.

Reply:
Maybe we could use std::vector<IdentityProjection> to specify how to project columns from filter input to operator output?

Reply:
> Wouldn't it be more convenient to map output columns to filter input columns?

I guess mapping filter input to output is more efficient since the mapping is mainly used in a loop of filter input column, so the index of column in the filter input is the key to search.

There are 3 places where the filter input -> output mapping is used:
1. When the filter input is built for the first time: 
```
foreach column in filter input
    if column in output, refer this column to output
    else create the column
```
2. When a batch of output is generated, the columns in filter input need to be dereferenced:
```
foreach reference column
    column_ptr.reset()
```
3. After generating a batch of output, prepare the filter input for the next batch:
```
foreach column in filter input
    if column in output, refer this column to the newly built output
    else if column is ArrayVector/MapVector, column.prepareForReuse()
```

Reply:
> Maybe we could use std::vector to specify how to project columns from filter input to operator output?

On vector vs map, performance-wise: 
- For case 1 & 3, they should have similar performance (Both are `O(1)` for one column, vector should be slightly faster).
- For case 2, vector is `O(number of column in filter input)`, map is `O(number of column that references to output in filter input)`. map wins if there are lots of columns in filter input but only a few are also in output.

Memory-wise, vector is `O(number of column in filter input)`, and map depends. I think vector will win if there are only a few columns in filter input since the map has overhead.

In all, I think there are trade-offs between vector and map and it depends on the data scheme to decide which is better. Maybe we can add a benchmark to MergeJoin?

Reply:
I don't think this performance matters here, just looking for ways to make the code a bit more readable. Thanks.